delete etcd socket file for unit tests

CaoShuFeng opened the pull request:
This change clean up the environment for etcd3 unit test.
Without this change, "make test" will leave some socket files in
workspace. And these socket files make hack/verify-generated-protobuf.sh
fails.

**Release note**:
<!--  Steps to write your release note:
1. Use the release-note-* labels to set the release note state (if you have access) 
2. Enter your extended release note in the below block; leaving it blank means using the PR title as the release note. If no release note is required, just write `NONE`. 
-->
```NONE
```

k8s-reviewable commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.kubernetes.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/43737)
<!-- Reviewable:end -->

CaoShuFeng commented:
This is the steps to reproduce the issue:
http://paste.openstack.org/show/604436/

CaoShuFeng committed:
delete etcd socket file for unit tests

This change clean up the environment for etcd3 unit test.
Without this change, "make test" will leave some socket files in
workspace. And these socket files make hack/verify-generated-protobuf.sh
fails.

timothysc commented:
@CaoShuFeng thx for the patch. 

/lgtm

CaoShuFeng commented:
@k8s-bot node e2e test this

timothysc commented:
@k8s-bot node e2e test this
